packages/flutter/lib/src/widgets/editable_text.dart

```diff
@@ -1074,9 +1086,15 @@ class EditableText extends StatefulWidget {
   /// {@template flutter.widgets.editableText.inputFormatters}
   /// Optional input validation and formatting overrides.
   ///
-  /// Formatters are run in the provided order when the text input changes. When
-  /// this parameter changes, the new formatters will not be applied until the
-  /// next time the user inserts or deletes text.
+  /// Formatters are run in the provided order when the text input is changed by
+  /// the user, or the composing text is committed.
+  ///
+  /// Typically, text modification should only be applied when text is being
+  /// committed by the IME and not on text under composition (i.e., only when
+  /// [TextEditingValue.composing] is collapsed).
+  ///
+  /// When this parameter changes, the new formatters will not be applied until
+  /// the next time the user inserts or deletes text.
```

Comment:
Does [MaxLengthEnforcement](https://github.com/flutter/flutter/pull/68086) affect this?

Reply:
Are you referring to the second paragraph? Unfortunately some IMEs (shimeji, for instance) restore the composing text on the next keypress after we format its composing text. It should be fine now that we restart the IME on these occasions, but it comes with a performance cost so we don't want to do the restart too frequently, thus the recommendation.